tests/unit/htdocs/hardening/RequestHashPinningTest.php

```
@@ -0,0 +1,240 @@
+                $inDoubleQuote = true;
+            } elseif ($char === '(') {
+                $depth++;
+            } elseif ($char === ')') {
+                $depth--;
+                if ($depth === 0) {
+                    return substr($line, $startPos + 1, $i - $startPos - 1);
+                }
+            }
+        }
+        return null; // unbalanced
+    }
```

PHPMD (2.15.0)

[warning] The method extractBalancedArgs() has a Cyclomatic Complexity of 15. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)